lib/core/external/date_service.dart

```diff
@@ -1,4 +1,5 @@
 class DateService {
-  int currentWeekday() => DateTime.now().weekday;
```

Comment:
Does not belong in a feature branch

Reply:
@fremartini I agree with leaving the codebase better than you found it, but consider the point of view of a reviewer. This branch's main purpose, according to the title, is to bring new functionality to the app. The reviewer would ideally only review the lines of code that directly drive this functionality.

Adding refactoring changes into the mix muddies the branch's purpose and makes the PR harder to review.  For instance, when you move files around, it is difficult to see if you only *really* changed one or two lines in the file, or if a new bug was introduced somewhere else in the code. This is because all the lines show as "added" lines in the diff.

If you believe refactoring is necessary (I don't necessarily disagree with the refactoring made in this branch), it's best to separate those changes into a dedicated branch pointing towards this one. Then, when the feature PR is merged, the refactoring PR can be merged afterwards.

This will simplify the review process by allowing reviewers to look at only one aspect of code changes at a time. Maybe someone has strong opinion about code style, but not the feature itself, or vice versa.

Reply:
@marfavi I understand it complicates the process of reviewing the code. I do personally not think this is as big an issue as you point out as booting up an editor and examining the diffs can be done rather quickly. Knowing the rate of which pull requests gets reviewed in this repository i do not believe the extra overhead of additional pull requests is worth practicing. 

On another note: i acknowledge my pull requests have lately included much refactoring which i know is unpleasant to the reviewer. I believe the restructure-the-app kinda pull requests that remain are near zero as i personally am content with the current structure we have achieved. This restructuring have in my opinion been a consequence of the entire team having to learn Flutter from scratch and how to best structure the app resulting in many try-this-architecture-out-and-see-how-it-works adventures, considering out limited experience designing software systems and apps.